module-2/backend/week_two.md

```diff
@@ -5,50 +5,68 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 1. At a high level, what is ActiveRecord? What does it do/allow you to do?
-2. Assume you have the following model:
 
-```ruby
-class Team << ActiveRecord::Base
-end
-```
+It is the inbetween for a database and a server. It is the ORM  that communicates/works with a SQL program. 
 
-What are some methods you can call on `Team`? If these methods aren't defined in the class, how do you have access to them?
+2. What kind of methods are `belongs_to`, and `has_many`? (i.e. class or instance) Give an example.
+They are Active record class methods. A car-dealership has_many cars. A Car belongs_to one car dealership.
 
-3. Assume that in your database, a team has the following attributes: "id", "name", owner_id". How would you find the name of a team with an id of 4? Assuming your class only included the code from question 2, how could you find the owner of the same team?
+3. What do they allow you to do?
 
-4. Assume that you added a line to your `Team` class as follows:
+They allow you to make relationships between databases.
 
-```ruby
-class Team << ActiveRecord::Base
-  belongs_to :owner
-end
-```
+4. What's the difference between agile workflow and waterfall method?
 
-Now how would you find the owner of the team with an id of 4?
+Agile workflow is all about building one full functionality at a time. All the way through, including the styling. Waterfall builds all aspects from the ground up at the same time.
+
+5. What is the difference between `#new` and `#create`?
+
+The new method needs to be saved. Create method does that automatically.
+
+6. At a basic level, what does cURL allow you to do?
+
+Had to look it up because I couldn't really remember. "cURL is a command line tool for getting or sending files using URL syntax"
 
-3. What do they allow you to do?
 7. In a database that's holding students and teachers, what will be the relationship between students and teachers? Draw the schema diagram.
+
+They would probably have an intermediary table called 'students_teachers'. This table would have a student_id and a teacher_id because a teacher has_many students and a student has_many teachers.
+
 8. Define foreign key, primary key, and schema.
+
+The foreign key is the variable in a table that connects one table to another. For instance, In the car/dealership example from above the dealership_id in the car table is a foreign key whereas the car's id is considered a primary key. The schema is an rb file that shows the relationships between all tables. It also includes the name/type of every column in a table.
+
 9. Describe the relationship between a foreign key on one table and a primary key on another table.
+They are the same value, but there positions in the relationships are different.
+
 10. What are the parts of an HTTP response?
+VERB PATH PROTOCOL headers, body
```

Comment:
This looks like a reqest to me (the VERB piece in particular). A response will have headers and a body, and the status (e.g. 200, 404, etc.) will be the first header line.